From 06de9882557745d20c062cf87c650568c61fc50d Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Tue, 28 Apr 2015 10:55:15 +0200
Subject: [PATCH] Unset frame listener before tearing down GLThreadManager.

Otherwise might lead to messages being delivered to the then dead
mGLHandlerThread via RequestThreadManager.mPreviewCallback ->
GLThreadManager.queueNewFrame().

diff --git a/core/java/android/hardware/camera2/legacy/RequestThreadManager.java b/core/java/android/hardware/camera2/legacy/RequestThreadManager.java
index f1f2f0c..202e6d1 100644
--- a/core/java/android/hardware/camera2/legacy/RequestThreadManager.java
+++ b/core/java/android/hardware/camera2/legacy/RequestThreadManager.java
@@ -963,6 +963,9 @@ public boolean handleMessage(Message msg) {
                         mDeviceState.setError(
                                 CameraDeviceImpl.CameraDeviceCallbacks.ERROR_CAMERA_DEVICE);
                     }
+                    if (mPreviewTexture != null) {
+                        mPreviewTexture.setOnFrameAvailableListener(null);
+                    }
                     if (mGLThreadManager != null) {
                         mGLThreadManager.quit();
                         mGLThreadManager = null;
